### PR TITLE
journalctl JSON rework

### PR DESCRIPTION
**Pull request by poettering**
this adds color and json-seq support among other fixes.

This is based on the JSON stuff that makes up #10316, hence review that one before this one.

### CODE REVIEW

File: src/shared/logs-show.c [hunk 1]
```diff
@@ -747,6 +748,96 @@ void json_escape(
...
+        d = hashmap_get(h, name);
+        if (d) {
+                struct json_data *w;
+
+                w = realloc(d, offsetof(struct json_data, values) + sizeof(JsonVariant*) * (d->n_values + 1));
+                if (!w)
+                        return log_oom();
+
+                d = w;
+                assert(hashmap_update(h, json_variant_string(d->name), d) >= 0);
```
Comment (keszybz) [inline src/shared/logs-show.c]: `assert_se`!

Comment (poettering) [reply src/shared/logs-show.c]: ouch

**Comment by poettering** [comment]
> It seems that slashes are "escaped" unnecessarily.

This appears to be how things are done usually... Note json.org for example, which lists the slash to escape...

https://stackoverflow.com/questions/1580647/json-why-are-forward-slashes-escaped

Not sure what the better approach there is: to escape or not to escape...

But i figure that choice is separate from this PR, as it's specific to json.c...

**Comment by poettering** [comment]
force pushed a new version. only change is the assert_se thing. the escaping of "/" is left as it is. ptal

**Pull request merged**